state/singular.go

```diff
@@ -54,7 +54,8 @@ func (s singularSecretary) CheckDuration(duration time.Duration) error {
 // SingularClaimer returns a lease.Claimer representing the exclusive right to
 // manage the model.
 func (st *State) SingularClaimer() corelease.Claimer {
-	return lazyLeaseManager{func() *lease.Manager {
-		return st.workers.singularManager()
+	return lazyLeaseManager{func() (lease.CheckerClaimer, error) {
```

Comment:
I think Bind() needs to be a private method. We just need to expose Claimer() and Checker().
So here we'd just need to access the claim as that's all we're returning:

return lazyLeaseManager{func() (lease.Claimer, error) {
    manager := st.workers.singularManager()
    return manager.Claimer(singularControllerNamespace, st.modelUUID())
}

Reply:
Yeah, you're right - also it's kind of an awkward name. I was thinking about doing that. I'll need to split `lazyLeaseManager` into `lazyLeaseClaimer` and `lazyLeaseChecker`, but that's ok.